ext/yarp/extconf.rb

```diff
@@ -19,32 +21,59 @@ def configure_debug
       end
 
       def configure_rubyparser
-        find_header("yarp.h", include_dir)
+        # unless File.exist?(ast_h_path = File.join(include_dir, "yarp", "ast.h"))
+        #   raise "yarp/ast.h is required"
+        # end
+        # FileUtils.touch(ast_h_path)
+
+        find_header("yarp.h", include_dir) or raise "yarp.h is required"
 
         # Explicitly look for the extension header in the parent directory
         # because we want to consistently look for yarp/extension.h in our
         # source files to line up with our mirroring in CRuby.
-        find_header("yarp/extension.h", File.join(__dir__, ".."))
+        find_header("yarp/extension.h", File.join(__dir__, "..")) or raise "yarp/extension.h is required"
 
         if static_link?
           static_archive_path = File.join(build_dir, "librubyparser.a")
           unless File.exist?(static_archive_path)
-            raise "Please run make to build librubyparser.a"
+            build_static_rubyparser
           end
           append_ldflags(static_archive_path)
         else
           unless find_library("rubyparser", "yp_parser_init", build_dir)
-            raise "Please run make to build librubyparser.so"
+            build_shared_rubyparser
+            unless find_library("rubyparser", "yp_parser_init", build_dir)
+              raise "could not build librubyparser.so"
+            end
```

Comment:
Are we ever going to use this code path?  I'd like to change `RUBY_FUNC_EXPORTED` such that it's empty _unless_ we're building `librubyparser`.  Currently Ruby is leaking YARP symbols, and if you try to compile and use YARP with Ruby head, it'll end up using the functions defined in Ruby rather than the ones in here.

Basically I want to do [this](https://github.com/tenderlove/ruby/commit/32101689902d233f062fb31cb7b0a34ee4136286), then add `-D YP_EXPORTED_FUNCTION=extern` to the CFLAGS when building the `.so`

Reply:
I expect you're right that we don't need it and that would simplify some of the hoops I jumped through in https://github.com/ruby/yarp/pull/1069

Can we leave that for a separate PR, though?

Reply:
Let's keep this branch for now, but I think we need to revisit, document, and most importantly test all YARP build possibilities.

For the gem, JRuby will need the shared object along with an FFI integration.  I'm not sure if that's true for Truffle Ruby.  I'd like it if CRuby didn't need to use the serialization interface in the gem, but we probably should test performance.  IOW if all Ruby implementations used the same interface, that would make our lives easier.  I just don't want it to be at the cost of performance on CRuby.

Reply:
I think for the Ruby API on TruffleRuby the deserialization will be much better than running through the C extension, but I need to benchmark. I plan to make a PR to implement `YARP.load` with Fiddle (or FFI) on JRuby/TruffleRuby.
I would prefer if CRuby would use that too, but indeed if the performance is significantly worse then we'll have to keep both ways (they used to be pretty close last time I looked at it for CRuby).

Reply:
> I plan to make a PR to implement `YARP.parse` with Fiddle (or FFI) on JRuby/TruffleRuby.

Early draft PR for that: https://github.com/ruby/yarp/pull/1112

Reply:
> Currently Ruby is leaking YARP symbols, and if you try to compile and use YARP with Ruby head, it'll end up using the functions defined in Ruby rather than the ones in here.

How does that happen? Does CRuby use the static library? That one doesn't export anything, so that should work fine.